File: ruby/spec/home_spec.rb
require_relative "spec_helper"

describe "home page" do
  before(:all)  { @s = new_session }
  before(:each) { @s.home_page.visit }

  let(:s) { @s }

  it "displays the Senior Full Stack Engineer position by default" do
    expect(s.jobs_panel.eng_job).to be_present
  end

  it "displays the QA Engineer position when the QA Engineer button is clicked" do
    s.jobs_panel.qa_button.click
    expect(s.jobs_panel.qa_job.wait_until(&:present?)).to be_present
  end
end

File: javascript/test/specs/home.spec.js
var HomePage = require('../ui/home.page');
var assert = require('assert');

// import HomePage from '../ui/home.page'

describe('Home page', function () {
  it('Displays the Senior Full Stack Engineer position by default', function () {
      HomePage.open();
      assert.ok(HomePage.jobs_panel.getText().includes('Are you an awesome developer'))
  });

  it('Displays the QA Engineer position when the QA Engineer button is clicked', function () {
      HomePage.open();
      HomePage.qa_button.click();
      assert.ok(HomePage.jobs_panel.getText().includes('Do you love high quality software'))
  });
});
